docs/CODE-REVIEW-GUIDELINES.md

```diff
@@ -0,0 +1,85 @@
+  - If you're not sure how someone wants their code reviewed or how they prefer to receive feedback, talk to them beforehand
+  - read the CDS [Code of Conduct](https://github.com/cds-snc/.github/blob/master/CODE_OF_CONDUCT.md)
+- Point out when something is particularly good or clever
+  - Reviews often focus on negative things so highlight good things as well when you see them
+- Be explicit with feedback
+  - If screenshots or gifs will help make your point, include them
+- Programming is subjective
+  - Different approaches are valid
+  - Ask for clarifications if you need any
+- Communicate which ideas you feel strongly about and those you don't
+- Never review code on weekends
+
+## Technical code review checklist
+
+### Logic
+
+- Does the code work?
+  - Does it perform its intended function?
+  - Is the logic correct?
+  - Is the proposed UI accessible?
+  - Make sure to test the working code yourself
+- Is all the code easily understood?
+  - Which parts are confusing to you and why?
+  - Multiple (nested) if/else blocks are code smells for complex logic
+  - If you are taking a lot of time to understand the code, perhaps it needs comments or refactoring
+- Can the code be simplified?
+  - Remove unused parameters or "generic" code that is only used once
+  - Don't solve problems you don't have
+- Does similar functionality already exist in the codebase?
+  - If so, why isn’t this functionality reused? (Note: it's okay to not reuse things — but it's worth asking the question.)
+- Can any of the code be replaced with library or built-in functions?
+- Does this change introduce any security concerns?
+  - Does it accept or handle user data?
+
+### Comments
+
+- Has all commented-out code been deleted?
+  - We can find past code using version control if needed
+- Are all functions commented?
+- Do comments exist to describe the intent of any thorny logic?
+- Are there comments that no longer make sense?
+
+### Errors
+
+- Can you think of any use case in which the code does not behave as intended?
+  - Run the code and try to cause errors
+- Are potential errors being caught and logged?
+- Are third-party utilities are used which may return errors?
+- Is any unusual behavior or edge-case handling described?
+
+### Tests
+
+- Do tests exist, and are they comprehensive?
+  - Do tests exist for errors or bad inputs?
+- Are tests checking things at the right level?
+  - Often, unit tests aren't suitable for complex flows
+- Do the existing set of tests pass?
+  - Tests should be running as part of a continuous integration workflow
+
+### Documentation
+
+- Does this change require an update to the README or configuration files?
+- Is the reason for the change outlined in the commit message?
+  - [A good commit message](https://github.com/alphagov/styleguides/blob/master/git.md) explains what changed and _why_ it was done this way
```

Comment:
This would be something I'd personally need to work on, but agree with. VS Code for some reason tries to get scary with you if your commit message is over 50 characters in a single line (and I only just realized that this is only talking about a single line and not the message as a whole), which has influenced me towards brevity in the commit message with longer PR descriptions.

Having reasons be documented in the commit message is definitely worth doing.

Reply:
Writing good commits is important! Basically what you should be thinking is that if someone comes along in a year to figure out why a change was made, the commit should explain it. (Some changes are very obvious, but not all.)

But I agree that unless you know pretty well how VIM works, the tools to do it are often pretty bad.